src/periphery/CurveLpPpo.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.18;
+
+import {ICurvePool} from "../interfaces/ICurvePool.sol";
+
+import {BaseOracle, IChainlinkOracle} from "./BaseOracle.sol";
+
+contract CurveLpPpo is BaseOracle {
+
+    string private _DESCRIPTION;
+
+    ICurvePool public immutable LP;
+    IChainlinkOracle public immutable COIN1_ORACLE;
+    IChainlinkOracle public immutable COIN2_ORACLE;
+
+    constructor(address _curvePool, address _coin1Oracle, address _coin2Oracle) {
+        LP = ICurvePool(_curvePool);
+        COIN1_ORACLE = IChainlinkOracle(_coin1Oracle);
+        COIN2_ORACLE = IChainlinkOracle(_coin2Oracle);
+        _DESCRIPTION = string(abi.encodePacked(LP.name(), " / ETH"));
+    }
+
+    function description() external view override returns (string memory) {
+        return _DESCRIPTION;
+    }
+
+    function latestRoundData()
+        public
+        view
+        override
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        (
+            uint80 roundId,
+            int256 ethPriceCoin1,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        ) = COIN1_ORACLE.latestRoundData();
+
+        (
+            uint80 roundIdCoin2,
+            int256 ethPriceCoin2,
+            uint256 startedAtCoin2,
+            uint256 updatedAtCoin2,
+            uint80 answeredInRoundCoin2
+        ) = COIN2_ORACLE.latestRoundData();
+
+        int256 minLpEthPrice = ethPriceCoin1 < ethPriceCoin2 ?
```

Comment:
it's not clear to me how this approach does not calculate a lower value after then ynethx vault allocates 


i get that it's manipulation resistant but i suppose it will cause a drop in value after depositing to this strategy

how would that work?

Reply:
current numbers are roughly :

 price_yneth = 1015767135652051362
 
 price_steth =  1189417475438185285
 
 virtualprice = 1000485588611310439